gnome-initial-setup/pages/branding-welcome/gis-branding-welcome-page.c

```diff
@@ -0,0 +1,289 @@
+  if (priv->description != NULL) {
+    opt_widget = WID("branding-text");
+    gtk_label_set_label (GTK_LABEL (opt_widget), priv->description);
+    gtk_widget_set_visible (opt_widget, TRUE);
+  }
+
+  if (priv->logo_path != NULL) {
+    opt_widget = WID("branding-logo");
+    gtk_image_set_from_file (GTK_IMAGE (opt_widget), priv->logo_path);
+    gtk_widget_set_visible (opt_widget, TRUE);
+  }
+}
+
+static void
+update_line_height_for_description (GisBrandingWelcomePage *page)
+{
+  GtkWidget *widget = NULL;
+  PangoAttrList *attr_list = NULL;
+  PangoAttribute *attr = NULL;
+
+  /* We can't set CSS's line-height property in GTK+ so we have to
+   * try to get a similar result changing the baseline with pango. */
+  attr_list = pango_attr_list_new ();
+
+  /* Rising the baseline of the 14px font (see branding-welcome-page.css)
+   * by 6px gives us a 20px line height. Note that there should be a way
+   * to get this value programatically, but all the experiments I did with
+   * Pango failed so far for some reason, so I'm hardcoding this for now. */
+  attr = pango_attr_rise_new (6 * PANGO_SCALE);
+
+  pango_attr_list_insert (attr_list,attr);
+  gtk_label_set_attributes (GTK_LABEL (WID ("branding-text")), attr_list);
+  pango_attr_list_unref (attr_list);
+}
+
+static void
+gis_branding_welcome_page_constructed (GObject *object)
+{
+  GisBrandingWelcomePage *page = GIS_BRANDING_WELCOME_PAGE (object);
+  GisBrandingWelcomePagePrivate *priv = gis_branding_welcome_page_get_instance_private (page);
+
+  G_OBJECT_CLASS (gis_branding_welcome_page_parent_class)->constructed (object);
+
+  gtk_container_add (GTK_CONTAINER (page), WID ("branding-welcome-page"));
+
+  update_branding_specific_info (page);
+  update_line_height_for_description (page);
+  load_css_overrides (page);
+
+  gis_page_set_complete (GIS_PAGE (page), TRUE);
+  gtk_widget_show (GTK_WIDGET (page));
+}
+
+static void
+gis_branding_welcome_page_finalize (GObject *object)
+{
+  GisBrandingWelcomePage *page = GIS_BRANDING_WELCOME_PAGE (object);
+  GisBrandingWelcomePagePrivate *priv = gis_branding_welcome_page_get_instance_private (page);
+
+  g_free (priv->title);
+  g_free (priv->description);
+  g_free (priv->logo_path);
+  g_free (priv->skip_pages);
+
+  G_OBJECT_CLASS (gis_branding_welcome_page_parent_class)->finalize (object);
+}
+
+static void
+gis_branding_welcome_page_class_init (GisBrandingWelcomePageClass *klass)
+{
+  GisPageClass *page_class = GIS_PAGE_CLASS (klass);
+  GObjectClass *object_class = G_OBJECT_CLASS (klass);
+
+  page_class->page_id = PAGE_ID;
+  object_class->constructed = gis_branding_welcome_page_constructed;
+  object_class->finalize = gis_branding_welcome_page_finalize;
+}
+
+static void
+gis_branding_welcome_page_init (GisBrandingWelcomePage *page)
+{
+  GisBrandingWelcomePagePrivate *priv = gis_branding_welcome_page_get_instance_private (page);
+
+  priv->title = NULL;
+  priv->description = NULL;
+  priv->logo_path = NULL;
+  priv->skip_pages = NULL;
```

Comment:
There's no need to initialize these to NULL -- GObject will do that for the whole private structure automatically.